### PR TITLE
Add PR Template

### PR DESCRIPTION
**Pull request by jestapinski**
This PR adds a PR template which will be used to auto-populate this description field. I added some basic things we seem to include on every PR as well as some additional points of clarification if we need reviews from outside folks.

As always, open to any thoughts on what we can add/remove!

### CODE REVIEW

**Comment by jestapinski** [comment]
This will look like https://github.com/mongodb/devhub/blob/38f996ad79bec972516698951cadeb9102f634e2/pull_request_template.md when done

**Comment by sophiali23** [comment]
this is great!! how about adding a `Testing Instructions` section so reviewers know what steps to take to test the PR and what the expected output should be?

**Comment by jestapinski** [comment]
> this is great!! how about adding a `Testing Instructions` section so reviewers know what steps to take to test the PR and what the expected output should be?

Awesome idea! I'll put up something shortly

File: pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+## Links:
+
+-   JIRA Ticket
+-   Staging Link
+
+## Description:
+
+-   Description of what this PR does and perhaps some context if needed
+
+## Testing
```
Comment (jestapinski) [inline pull_request_template.md:10]: Hey @sophiali23 what do you think?

Comment (sophiali23) [reply pull_request_template.md:10]: love it!

**Review by sophiali23 — APPROVED** [review]
lgtm! 😀

**Review by harikakotipalli — APPROVED** [review]

**Pull request merged**